client/my-sites/checkout/checkout-thank-you/marketplace/marketplace-thank-you-theme-section.tsx

```diff
@@ -112,6 +112,13 @@ export const ThankYouThemeSection = ( { theme }: { theme: any } ) => {
 		[ siteId, theme ]
 	);
 
+	// Clear completed activated theme request state to avoid displaying the Thanks modal
+	useEffect( () => {
+		return () => {
+			dispatch( clearActivated( siteId ) );
+		};
+	}, [ dispatch, siteId ] );
```

Comment:
On  #75546 I added it to `useThemesThankYouData` which I think should live the logic for anything theme related on Thank You page. What do you think?

Reply:
I think `useThemesThankYouData` could be a better fit for this logic 👍 . So I will delete this piece from here and once https://github.com/Automattic/wp-calypso/pull/75546 is merged I can re-test